### ISSUE TITLE
[Search Archive API] Endpoint returns `503`s for all errored searches

### ISSUE THREAD

**Issue by hartator** [serpapi/public-roadmap#79]
Our endpoint currently returns `503`s for all errored searches. This can be an issue as `503`s can be interpreted as a server error with no content on our side when we do. It can be especially an issue with tool like `wget`:

![image](https://user-images.githubusercontent.com/307597/158547916-1362aba4-cc51-4e0e-8579-8ad462b59f3a.png)

Maybe we can return `200` in these cases. What do you guys think?

**Comment by aciddjus** [comment]
Yes, currently have an exact problem with Yelp API.

**Comment by aciddjus** [comment]
> Yes, currently having the exact problem with Yelp API.

**Comment by ritu1337** [comment]
For inspiration, I have some notes from web scraping on websites that return `503`'s:

Shopify (from before they used Cloudflare):
<img width="736" alt="CleanShot 2022-03-16 at 20 47 43@2x" src="https://user-images.githubusercontent.com/12263752/158677467-8120a6ed-5a18-4b6b-ba61-8e95ec6ba39c.png">

Varnish Cache:
`<!DOCTYPE html>\n<html>\n<head>\n<title>503 Backend fetch failed</title>\n</head>\n<body>\n<h1>Error 503 Backend fetch failed</h1>\n<p>Backend fetch failed</p>\n<h3>Guru Meditation:</h3>\n<p>XID: 653411604</p>\n<hr>\n<p>Varnish cache server</p>\n</body>\n</html>\n`

Squid:
`<p>The following error was encountered while trying to retrieve the URL: <a href="http://api.net-a-porter.com/MRP/US/en/detail/1048642">http://api.net-a-porter.com/MRP/US/en/detail/1048642</a></p>  <blockquote id="error"> <p><b>Connection to 104.92.20.58 failed.</b></p> </blockquote>  <p id="sysmsg">The system returned: <i>(113) No route to host</i></p>  <p>The remote host or network may be down. Please try the request again.</p>`

But I think that it's odd when an API or a website returns a [>=500](https://cloud.google.com/storage/docs/json_api/v1/status-codes#500_Internal_Server_Error) error with something more descriptive than just a generic "Internal Server Error" message. I think that if we return a valid JSON then we should either return a `200` or we could think of a fitting `>=403` status?

**Comment by hartator** [comment]
Yes, I guess `503` (or other `500`errors) is definitely misleading when we do return a valid JSON.

> Yes, currently having the exact problem with Yelp API.

What do you mean? It should not be depend on the engine.

**Comment by aciddjus** [comment]
> What do you mean? It should not be depend on the engine.

Yes, my mistake for mentioning Yelp in particular. Just the thing I'm currently working on.

**Comment by hilmanski** [comment]
Was this report for an "empty result" search? If so, can we close this?

**Comment by schaferyan** [comment]
I think we can close this since we haven't had any reports since 2022
